### PR TITLE
Question model and tests

### PR DESCRIPTION
**Pull request by carolinedanzi**
### Description
Updated the Question model to match the question definition.

Paired with: @AWBaumann 

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Approvals
Want approval from any one reviewer

### CODE REVIEW

File: universal-application-tool-0.0.1/app/models/Question.java [hunk 1]
```diff
@@ -1,34 +1,77 @@
...
+  private @Constraints.Required String name;
+
+  private @Constraints.Required String description;
+
+  private @Constraints.Required @DbJsonB ImmutableMap<Locale, String> questionText;
+
+  private ImmutableMap<Locale, String> questionHelpText;
+
+  public Question(QuestionDefinition questionDefinition) {
+    this.questionDefinition = questionDefinition;
```
Comment (bion) [inline universal-application-tool-0.0.1/app/models/Question.java:37]: nit: 
```suggestion
    this.questionDefinition = checkNotNull(questionDefinition);
```

File: universal-application-tool-0.0.1/conf/evolutions/default/3.sql [hunk 2]
```diff
@@ -9,7 +9,12 @@ create table if not exists applicants (
 
 create table if not exists questions (
   id bigserial primary key,
-  object jsonb not null
+  version bigserial,
+  path varchar,
+  name varchar,
+  description varchar,
+  question_text jsonb,
+  question_help_text jsonb
```
Comment (bion) [inline universal-application-tool-0.0.1/conf/evolutions/default/3.sql:17]: question (non-blocking): what's the plan for persisting the question's question type?

**Review by bion — APPROVED** [review]

File: universal-application-tool-0.0.1/app/models/Question.java (hunk 1 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/models/Question.java]: Done!

File: universal-application-tool-0.0.1/app/models/Question.java [hunk 3]
```diff
@@ -1,34 +1,77 @@
...
+  /** Populates column values from {@link QuestionDefinition}. */
+  @PreUpdate
+  public void persistChangesToQuestionDefinition() {
+    this.id = questionDefinition.getId();
+    this.version = questionDefinition.getVersion();
+    this.path = questionDefinition.getPath();
+    this.name = questionDefinition.getName();
+    this.description = questionDefinition.getDescription();
+    this.questionText = questionDefinition.getQuestionText();
+    this.questionHelpText = questionDefinition.getQuestionHelpText().orElse(null);
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/models/Question.java:55]: If this was `orElse(ImmutableMap.of())`, would `loadQuestionDefinition` be able to use `Optional.of(this.questionHelpText)` and we could avoid nulls altogether?

File: universal-application-tool-0.0.1/test/models/QuestionTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    QuestionRepository repo = app.injector().instanceOf(QuestionRepository.class);
+    QuestionDefinition definition =
+        new QuestionDefinition(1L, 1L, "test", "my.path", "", ImmutableMap.of(), Optional.empty());
+    Question question = new Question(definition);
+
+    question.save();
+
+    Question found = repo.lookupQuestion(definition.getId()).toCompletableFuture().join().get();
+
+    assertThat(found.getQuestionDefinition().getId()).isEqualTo(definition.getId());
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/test/models/QuestionTest.java:26]: nit(non-blocking): I like to see the `.isEqualTo` values as raw values when possible.

Comment (kendrickt) [reply universal-application-tool-0.0.1/test/models/QuestionTest.java:26]: ditto for the `canSerializeLocalizationMaps` test case.

**Comment by kendrickt** [comment]
Is it weird that the `Program` model stores serialized `QuestionDefinition`s and not `Question`s?

File: universal-application-tool-0.0.1/app/models/Question.java (hunk 3 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/models/Question.java:55]: Agree that anytime we have null values is cause for concern, but I think we want nulls here - if a question doesn't have help text, we want `Optional.empty()`, not an `Optional` with an empty map

File: universal-application-tool-0.0.1/test/models/QuestionTest.java (hunk 4 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/test/models/QuestionTest.java:26]: Sounds good to me! Will update tests

File: universal-application-tool-0.0.1/conf/evolutions/default/3.sql (hunk 2 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/conf/evolutions/default/3.sql:17]: We're working on it - we need to persist the QuestionType enum safely, then reform the question into the proper class on load. We're working on this today (Friday)

**Pull request merged**